src/clang-c-frontend/clang_c_convert.cpp

```diff
@@ -1871,13 +1885,17 @@ bool clang_c_convertert::get_expr(const clang::Stmt &stmt, exprt &new_expr)
           return true;
 
         typet elem_type;
-        if(t.is_struct())
+        if(t.is_struct() || t.is_union())
           elem_type = to_struct_union_type(t).components()[i].type();
```

Comment:
remove added whitespaces below

Reply:
I am not sure what you mean here. Are you referring to the whitespaces between "if" and "else"?

Reply:
yes